src/sql/workbench/common/editor/query/queryEditorInput.ts

```diff
@@ -346,6 +321,28 @@ export abstract class QueryEditorInput extends EditorInput implements IConnectab
 			}
 		}
 		this._onDidChangeLabel.fire();
+
+		this.contextualizeEditorForCopilot();
```

Comment:
Did you verify that this is called when the connection is changed as well?

Reply:
This is also called when the query editor connection is changed to a different server.